modules/Microsoft.Network/vpnGateways/.test/common/deploy.test.bicep

```diff
@@ -0,0 +1,83 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.network.vpngateways-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nvgcom'
+
+// =========== //
+// Deployments //
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-paramNested'
+  params: {
+    virtualHubName: 'dep-<<namePrefix>>-vh-${serviceShort}'
+    virtualWANName: 'dep-<<namePrefix>>-vw-${serviceShort}'
+    vpnSiteName: 'dep-<<namePrefix>>-vs-${serviceShort}'
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    virtualHubResourceId: resourceGroupResources.outputs.virtualHubResourceId
+    bgpSettings: {
+      asn: 65515
+      peerWeight: 0
+    }
+    connections: [
+      {
+        connectionBandwidth: 100
+        enableBgp: false
+        name: 'Connection-dep-<<namePrefix>>-vs-${serviceShort}'
```

Comment:
Is the connection name intended to be like `'Connection-<<vpnSiteName>>'`? In that case I'd suggest to create a variable for the vpnSiteName and use that here and as `resourceGroupResources` input